lib/broccoli/ember-app.js

```diff
@@ -168,6 +168,16 @@ class EmberApp {
 
     this._isPackageHookSupplied = typeof this.options.package === 'function';
     this._cachedAddonBundles = {};
+
+    if (this.project.perBundleAddonCache && this.project.perBundleAddonCache.numProxies > 0) {
+      if (this.options.addons.whitelist && this.options.addons.whitelist.length) {
+        throw new Error('[ember-cli] addon bundle caching is disabled for apps that specify an addon `whitelist`');
```

Comment:
Hmm, I think we need to beef up these error messages. It may not be obvious which addons have opt-ed into bundle caching, can we include the list in the message?